gunicorn/arbiter.py

```diff
@@ -186,7 +186,6 @@ def init_signals(self):
         # initialize all signals
         for s in self.SIGNALS:
             signal.signal(s, self.signal)
-        signal.signal(signal.SIGCHLD, self.handle_chld)
```

Comment:
the reason we want to have the CGHLD signal handle separately is that we don't want to queue on it and ignore it  or having to reinstall it.  afaik originally it was just waking the arbiter to do asynchronous reaping. Maybe we should remove instead the `reap_workers` call there and let it async? That way we maximise the change to not loose a signal in the queue, since it's processed before.

Reply:
I'm re-reading the links I posted and I don't think we need to reinstall the handler. I think Python will do it automatically because it's safe on all platforms we support (it won't lead to infinite loop). That is why we never needed to reinstall it before, and that should not change. I can remove that part. But if we want to be extra safe, I can keep the reinstall, but it is only safe to do on those platforms _after_ reaping.

> Maybe we should remove instead the reap_workers call there and let it async?

That's what I made this PR to do. The current way we handle that is with the signal queue and a wakeup. If we are afraid of losing the signal, we can either have a separate flag for CHLD or always append the CHLD signal no matter how long the queue is.